cli/derive/src/lib.rs

```diff
@@ -0,0 +1,168 @@
+/// generate_endpoints!(3, my_endpoint: 2, 4, anotherOne: 5, );
+/// ```
+#[proc_macro]
+pub fn generate_endpoints(input: TokenStream) -> TokenStream {
+    let EndpointList(list) = parse_macro_input!(input as EndpointList);
+    let arg_names = (1_u8..).map(|count| {
+        Ident::new(
+            format!("__endpoint_arg_{count}").as_str(),
+            Span::call_site(),
+        )
+    });
+    let arg_types = (1_u8..).map(|count| {
+        Ident::new(
+            format!("__Endpoint_Arg_{count}").as_str(),
+            Span::call_site(),
+        )
+    });
+    let mut endpoints = Vec::new();
+
+    for item in list {
+        let (fun_name, arg_names, arg_types) = match item {
+            EndpointItem::ArgCount(arg_count) => {
+                let fun_name = Ident::new(&format!("endpoint{arg_count}"), Span::call_site());
+                #[allow(clippy::expect_used)]
+                let count = arg_count
+                    .base10_parse::<usize>()
+                    .expect("Already checked at parse stage");
+                let arg_names = arg_names.clone().take(count).collect::<Vec<_>>();
+                let arg_types = arg_types.clone().take(count).collect::<Vec<_>>();
+                (fun_name, arg_names, arg_types)
+            }
+            EndpointItem::NameAndArgCount {
+                name: fun_name,
+                arg_count,
+            } => {
+                #[allow(clippy::expect_used)]
+                let count = arg_count
+                    .base10_parse::<usize>()
+                    .expect("Already checked at parse stage");
+                let arg_names = arg_names.clone().take(count).collect::<Vec<_>>();
+                let arg_types = arg_types.clone().take(count).collect::<Vec<_>>();
+                (fun_name, arg_names, arg_types)
+            }
+        };
+
+        let expanded = quote! {
+            #[inline]
+            #[allow(clippy::redundant_pub_crate)]
+            pub(crate) fn #fun_name < O, E, F, Fut, Fil, #( #arg_types ),* > (
+                f: F,
+                router: Fil,
+            ) -> impl Filter<Extract = (WarpResult<O, E>,), Error = Rejection> + Clone
+            where
+                Fil: Filter<Extract = ( #( #arg_types ),* ), Error = Rejection> + Clone,
+                F: Fn( #( #arg_types ),* ) -> Fut + Copy + Send + Sync + 'static,
+                Fut: std::future::Future<Output = Result<O, E>> + Send,
+                #( #arg_types: Send ),*
+                {
+                    router.and_then(move | #( #arg_names ),* | async move {
+                        Ok::<_, Infallible>(WarpResult(f( #( #arg_names ),* ).await))
+                    })
+                }
+        };
+
+        endpoints.push(expanded);
+    }
+
+    quote! {
+        #( #endpoints )*
+    }
+    .into()
+}
+
+#[derive(Debug)]
+struct EndpointList(Vec<EndpointItem>);
+
+#[derive(Debug)]
+enum EndpointItem {
+    NameAndArgCount { name: Ident, arg_count: LitInt },
```

Comment:
would it be possible to have `arg_count` be `u8`? Since you're parsing it at the call site anyways